rfcs/0000-consistent-resource-names.md

```diff
@@ -0,0 +1,97 @@
+- Feature Name: project-name-rationalization
+- Start Date: 2022-05-16
+- RFC PR: [mbta/technology-docs#13](https://github.com/mbta/technology-docs/pull/13)
+- Asana task: [RFC: rationalizing how we name projects](https://app.asana.com/0/1113179098808463/1202191100361265/f)
+- Status: Proposed
+
+# Summary
+[summary]: #summary
+
+Resource names in CTD have a consistent pattern.
+
+# Motivation
+[motivation]: #motivation
+
+As CTD has grown, multiple teams have independently named their projects and the associated resources. As these names have been generated independently, they are not always consistent about how they use punctuation to separate multiple words in a resource name, leading to confusion.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Within CTD, we use one of the following patterns for naming projects and resources:
+- kebab-case: all lowercase, separated by hyphens
+- snake_case: all lowercase, separated by underscores
+- camelCase: first word is lower case, other words are Title case with no separation
+- PascalCase: all words are Title case with no separation
+
+We use the following patterns for resources:
+- GitHub Repositories: snake_case
+- Elixir modules: snake_case filenames, PascalCase module names (based on the filenames)
+	- https://hexdocs.pm/elixir/1.12.3/naming-conventions.html
+- Python modules: snake_case
+	- https://peps.python.org/pep-0008/
+- React components: PascalCase
+- TypeScript modules: camelCase
+- Shell scripts: kebab-case with no trailing `.sh`
+- Domain names: kebab-case
+	- https://www.ssl.com/faqs/underscores-not-allowed-in-domain-names/
+- S3 buckets: kebab-case
+	- https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
+- Terraform resources: kebab-case
+- Terraform modules: kebab-case
+- Terraform variables: snake_case
```

Comment:
@ianwestcott these (in particular modules) are based on our current practice. However. the larger Terraform ecosystem appears to be using snake_case. Do you have thoughts here?

Reply:
It does look like [Terraform's official recommendation is to use `_` (underscore) for all named things](https://www.terraform-best-practices.com/naming), which I was unaware of when I started writing out/importing our configuration. So I'm okay with adopting the official convention, even if it means a lot of what exists now is non-compliant.